src/cnfizers/Cnfizer.h

```diff
@@ -38,95 +38,79 @@ class SimpSMTSolver;
 class THandler;
 struct SMTConfig;
 
+
+
 //
 // Generic class for conversion into CNF
 //
 class Cnfizer
 {
 public:
-    SimpSMTSolver&      solver;
+    struct ClauseCallBack {
```

Comment:
What is the point of this structure? 
Only thing it does is an operator rewrite

Reply:
It's a callback, a customization point to specify, what should happen with the clauses that Cnfizer produces. This means that `Cnfizer` is only concerned with creating the clauses and does not need to care about specifics of what to do with the clauses.